openmc/mgxs/mgxs.py

```diff
@@ -2829,6 +2832,294 @@ def nu(self, nu):
+                                                   num_polar, num_azimuthal)
+        self._rxn_type = 'diffusion-coefficient'
+        cv.check_value('nu for diffusion-coefficient', nu, [False])
+
+    @property
+    def rxn_rate_tally(self):
+        if self._rxn_rate_tally is None:
+            # Switch EnergyoutFilter to EnergyFilter
+            p1_tally = self.tallies['scatter-1']
+            old_filt = p1_tally.filters[-2]
+            new_filt = openmc.EnergyFilter(old_filt.values)
+            p1_tally.filters[-2] = new_filt
+
+            # Slice Legendre expansion filter and change name of score
+            p1_tally = p1_tally.get_slice(filters=[openmc.LegendreFilter],
+                                          filter_bins=[('P1',)],
+                                          squeeze=True)
+            p1_tally._scores = ['scatter-1']
+
+            # Compute total cross section
+            total_xs = self.tallies['total'] / self.tallies['flux (tracklength)']
+
+            # Compute transport correction term
+            trans_corr = self.tallies['scatter-1'] / self.tallies['flux (analog)']
+
+            # Compute the diffusion coefficient
+            transport = total_xs - trans_corr
+            dif_coef = transport**(-1) / 3.0
+
+            self._rxn_rate_tally = dif_coef * self.tallies['flux (tracklength)']
+            self._rxn_rate_tally.sparse = self.sparse
+
+        return self._rxn_rate_tally
+
+    @property
+    def xs_tally(self):
+        if self._xs_tally is None:
+            if self.tallies is None:
+                msg = 'Unable to get xs_tally since tallies have ' \
+                      'not been loaded from a statepoint'
+                raise ValueError(msg)
+
+            self._xs_tally = self.rxn_rate_tally / self.tallies['flux (tracklength)']
+            self._compute_xs()
+
+        return self._xs_tally
+
+    def get_condensed_xs(self, coarse_groups, condense_dif_coef=True):
+        """Construct an energy-condensed version of this cross section.
+        Parameters
+        ----------
+        coarse_groups : openmc.mgxs.EnergyGroups
+            The coarse energy group structure of interest
+        Returns
+        -------
+        MGXS
+            A new MGXS condensed to the group structure of interest
+        """
+
+        cv.check_type('coarse_groups', coarse_groups, EnergyGroups)
+        cv.check_less_than('coarse groups', coarse_groups.num_groups,
+                           self.num_groups, equality=True)
+        cv.check_value('upper coarse energy', coarse_groups.group_edges[-1],
+                       [self.energy_groups.group_edges[-1]])
+        cv.check_value('lower coarse energy', coarse_groups.group_edges[0],
+                       [self.energy_groups.group_edges[0]])
+
+        # Clone this MGXS to initialize the condensed version
+        condensed_xs = copy.deepcopy(self)
+
+        if condense_dif_coef:
```

Comment:
Not sure I understand the need for this

Reply:
The idea of cloning the original MGXS is to initialize the condensed_xs with all the existing attributes. This is done also in the main MGXS class. 

As for the "if condense_dif_coef" I'm not sure either. Who would call get_condensed_xs for the DiffusionCoefficient class and set condense_dif_coef to False? I am comfortable removing that if-statement.

Reply:
I'd agree that we should just remove the ``if condense_dif_coef`` block. Seems like there is no need for that because of course one wants that.